### PR TITLE
Refactor to use HTTP client

### PR DESCRIPTION
**Pull request by respawned86**
When migrating to newer versions of .NET, we discovered that the old `HttpWebRequest` has very poor performance and should be replaced with the new `HttpClient`.

From our tests, we concluded that the old `HttpWebRequest` kept creating new sockets on the machine, instead of reusing them, which eventually meant that it ran out of sockets and became very slow.

Only 2 options could not be migrated, since they are not available with `HttpClient`:
- UseNagle, which has something to do with grouping requests
- Exception status code, which is possible in .NET 5

A "TODO" comment has been added for the last part, which can be enabled when migrating the code to .NET 5 or newer.

Unused functionality was removed, like the response header collection on scanner information and the credentials manager in the VNET web requester.

The previous functionality has been left in `GatewayWebRequester`, which we do not use.

Exceptions thrown in `VNetWebRequester` are now rethrown correctly, so they include the original call stack.

### CODE REVIEW

File: Microsoft.HBase.Client/Requester/VNetWebRequester.cs [hunk 1]
```diff
@@ -69,147 +74,105 @@ public Response IssueWebRequest(string endpoint, string query, string method, St
...
+                    Content = content,
+                    Headers =
                     {
-                        req = await httpWebRequest.GetRequestStreamAsync().WithTimeout(
-                                                    TimeSpan.FromMilliseconds(remainingTime),
-                                                    "Waiting for RequestStream");
 
-                        remainingTime = options.TimeoutMillis - watch.ElapsedMilliseconds;
-                        if (remainingTime <= 0)
+                        ConnectionClose = !options.KeepAlive,
```
Comment (Fdhansen) [inline Microsoft.HBase.Client/Requester/VNetWebRequester.cs:117]: Read up on this part. It seems correct that if options have KeepAlive false, that close should be true. 

If we do not apply different options the `RequestOptions.GetDefaultOptions()` will return `KeepAlive = true` and that is an important point, as setting ConnectionClose = true in cases where we do not want it, could be detrimental to the performance.

File: Microsoft.HBase.Client/Requester/VNetWebRequester.cs [hunk 2]
```diff
@@ -69,147 +74,105 @@ public Response IssueWebRequest(string endpoint, string query, string method, St
...
-                    }
-                    finally
-                    {
-                        if (req != null) {
-                            req.Close();
+                            MediaTypeWithQualityHeaderValue.Parse(_contentType)
                         }
                     }
+                };
+                var cts = new CancellationTokenSource(options.TimeoutMillis);
```
Comment (Fdhansen) [inline Microsoft.HBase.Client/Requester/VNetWebRequester.cs]: Any reason we are doing this outside the inner try-block? Seems to me like it is only used within the scope of the inner try-block, and should probably be declared in there as well?

**Review by Fdhansen — CHANGES_REQUESTED** [review]
There is an issue with the protobuf when loading recordings with domMutations. 

It throws an error when trying to load the recording as seen below 👍 
image

When stepping back in the callstack, the file paths attempted read could as the example below:
image

**Comment by Fdhansen** [comment]
Link to stackoverflow in exception:
https://stackoverflow.com/questions/2152978/using-protobuf-net-i-suddenly-got-an-exception-about-an-unknown-wire-type

**Comment by respawned86** [comment]
Nicely spotted @Fdhansen!

It appears that the DOM mutations in your example were not found and we did not handle HTTP 404: Not Found properly.

I have fixed it, so the behavior is the same as before migrating to `HttpClient`.

The previous code threw an exception on 4xx and 5xx responses, which is now done by calling EnsureSuccessStatusCode.

HttpRequestException does not have a status code before .NET 5, so it is added as exception data, but a TODO comment has been left to change this when migrating to .NET 5

**Review by Fdhansen — APPROVED** [review]

**Pull request merged**